mitosheet/src/components/taskpanes/Graph/GraphStyleTab.tsx

```diff
@@ -399,30 +973,101 @@ function GraphStyleTab(props: {
                     </>
                 }
             </CollapsibleSection>
+            <CollapsibleSection title='Facet Styling'>
+                {!props.userProfile.isPro &&
+                    <Row justify='space-between' align='center'>
+                        <p className='text-body-1'>
+                            Want to style facet plots? <a href='https://trymito.io/plans' target='_blank' rel="noreferrer"><span className='text-body-1-link'>Upgrade to Mito Pro.</span></a>
+                        </p>  
+                    </Row>
+                }
+                {props.userProfile.isPro &&
+                    <>
+                        <Row justify='space-between' align='center' title='The number of plots to display per row. Has no effect when facet row is used.'>
+                            <Col>
+                                <p>
+                                    Facet column wrap
+                                </p>
+                            </Col>
+                            <Input 
+                                value={graphCreationParams.facet_col_wrap ? graphCreationParams.facet_col_wrap.toString() : ''} 
+                                type='number'
+                                width='small'
+                                placeholder='num cols'
+                                onChange={(e) => {
+                                    const newNumCols = e.target.value === '' ? undefined : e.target.value
+                                    props.setGraphParams(prevGraphParams => {
+                                        const graphParamsCopy: GraphParams = JSON.parse(JSON.stringify(prevGraphParams)); 
+                                        return {
+                                            ...graphParamsCopy,
+                                            graphCreation: {
+                                                ...graphParamsCopy.graphCreation,
+                                                facet_col_wrap: newNumCols as number | undefined
```

Comment:
A reminder that there is a difference between _forcing a type_ and _casting the string to a number_. 

Consider the following:
```
const x = "string";
const y: number = x as number;
```
What is the value of `y`? Well, it's "string" still. All you've done is strong-armed the compiler into thinking the that it is a number, when it is in-fact, not. The `as` keyword doesn't change the data that's there, it just changes what type the compiler thinks it is.

Consider the next example.
```
const x = "string";
const y: number = parseFloat(x);
```
What is y equal to? y is NaN! This is because actually parsing the float fails in this case, as it isn't parsable.

# What you want to do here

In this case, we actually want to send a _number_ to the backend. Because you aren't actually parsing the string to a number, this is why you have to do that conversion on the backend. 

That being said, this can get really annoying if the user wants to be able to enter decimals. Check out the FillNaN taskpane for example (line 342). Because we want to allow the user to:
1. Enter a number.
2. That has decimals,
3. So might not be a number while it's being entered (e.g. "." is not a number, but ".123" is)
We need to store the value as a string, and then send it to the backend as a number, only by doing the final conversion _right before_ we send the number to the backend!

To deal with this, the best way I've figured out of generally handling things is:
1. Have a notion of FrontendGraphParams and BackendGraphParams. Frontend params store these values a strings, and backend values store them as numbers. (You could even make a parameteric type to represent this, so you only have to specify this once... but this might be more complexity than it's worth). 
2. Store FrontendGraphParams on the frontend. Convert to backendGraphParams before sending to the backend. This is just just right before the data is sent to the backend. 

Although this might be _net_ more lines of code, it's better because the backend gets data in a format it can directly operate on without casting things, and the frontend is just responsible for sending the correct types. And it accomplishes this while allowing users to enter in whatever number they set. 

# A note on this

This is a bit of work, and I'm happy to do it, but it also could be educational for you. Would love if you want to try it for a bit (an hour?) and then give up if you get stuck. I can finish it off, and we can discuss. 

This is the sort of thing that has popped up exactly _twice_ before, so writing standards before it ahead of time is really hard (it barely was even a distict thing in my head until I saw this casting going on), but I think that this is a great chance to align on these, figure out what a good way to handle them generally is, and learn for the future!

This isn't just me asking you to spin you're wheels for no reasons -- I think you'll learn a lot (about types, casting), and we can have a chance to talk about and figure out better ways to approach this going forward. Let me know if you have more thoughts here.

Reply:
Done! Thanks for this super thorough explanation! Very helpful.